### PR TITLE
Fix audit max backups and log location

### LINKED ISSUE #384: Fix apiserver audit logging
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**: The apiserver is logging to `/etc/kubernetes` and is also never deleting old backups

**What you expected to happen**: The apiserver should log to `/var/log/kubernetes` and should keep no backups

**How to reproduce it (as minimally and precisely as possible)**: `tarmak clusters apply` and observe audit logs in `/etc/kubernetes/audit`

### PR DESCRIPTION
**Pull request by dippynark**
**What this PR does / why we need it**: Fixes audit log max backups and log location

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #384

```release-note
Fix apiserver audit log location and rotation
```

### CODE REVIEW

**Comment by charlieegan3** [comment]
/lgtm

I've not tested but looks fine.

File: puppet/modules/kubernetes/manifests/apiserver.pp [hunk 1]
```diff
@@ -192,13 +192,18 @@
...
     file {$audit_log_directory:
-      ensure => directory,
-      mode   => '0750',
-      owner  => $::kubernetes::params::user,
-      group  => $::kubernetes::params::group,
+      ensure  => directory,
+      mode    => '0750',
+      owner   => $::kubernetes::params::user,
+      group   => $::kubernetes::params::group,
+      require => Kubernetes::Symlink['apiserver'],
```
Comment (simonswine) [inline puppet/modules/kubernetes/manifests/apiserver.pp]: This require seems unnecessary to me

File: puppet/modules/kubernetes/manifests/apiserver.pp [hunk 2]
```diff
@@ -2,9 +2,9 @@
 class kubernetes::apiserver(
   $allow_privileged = true,
   Optional[Boolean] $audit_enabled = undef,
-  String $audit_log_directory = '/etc/kubernetes/audit',
-  String $audit_log_path = '/etc/kubernetes/audit/kubernetes-audit.log',
-  String $audit_policy_file = '/etc/kubernetes/audit/audit-policy.yaml',
+  String $audit_log_directory = '/var/log/kubernetes',
```
Comment (simonswine) [inline puppet/modules/kubernetes/manifests/apiserver.pp]: This change needs to be also in fluent bit

Comment (simonswine) [inline puppet/modules/kubernetes/manifests/apiserver.pp]: This also might interfere with a default logrotate rule centos sets up. Worth checking how logrotate is setup by centos

Comment (dippynark) [reply puppet/modules/kubernetes/manifests/apiserver.pp]: Looks like logrotate doesn't touch general files in `/var/log`
```
$ cat /etc/logrotate.conf | grep '/var/log'
/var/log/wtmp {
/var/log/btmp {
[root@luke-cluster|master|ip-10-99-53-8 ~]$ cat /etc/logrotate.d/* | grep '/var/log'
/var/log/chrony/*.log {
/var/log/puppetlabs/pxp-agent/*.log {
/var/log/cron
/var/log/maillog
/var/log/messages
/var/log/secure
/var/log/spooler
/var/log/wpa_supplicant.log {
/var/log/yum.log {
```

Comment (simonswine) [reply puppet/modules/kubernetes/manifests/apiserver.pp]: Thanks

**Pull request merged**